src/DynamoCore/Graph/Annotations/AnnotationModel.cs

```diff
@@ -141,15 +176,28 @@ public IEnumerable<ModelBase> Nodes
             get { return nodes; }
             set
             {
-                nodes = value.ToList(); ;
+                // Before setting this property
+                // we need to reset all previous values.
+                if (nodes != null)
+                {
+                    foreach (var node in nodes)
+                    {
+                        node.BelongsToGroup = false;
+                    }
+                }
```

Comment:
For this kind of loop with only one action for each, we usually leverage `ForEach()`
```
nodes = value.ToList().ForEach(x => x.BelongsToGroup = false);
```